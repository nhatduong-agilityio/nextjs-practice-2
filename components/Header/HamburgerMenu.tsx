// Libs
import { CloseIcon } from '@chakra-ui/icons'
import { IconButton } from '@chakra-ui/react'
import React, { memo } from 'react'

// Components
import BurgerMenuIcon from '@components/Icons/BurgerMenuIcon'

interface IProps {
  toggleMenu: boolean
  onHandleToggleMenu: () => void
}

const HamburgerMenu = memo(({ toggleMenu, onHandleToggleMenu }: IProps) => (
  <IconButton
    data-testid='hamburgerMenu'
    style={{
      marginLeft: '18px',
    }}
    variant='ghost'
    aria-label='Open Menu'
    size='sm'
    icon={
      toggleMenu ? (
        <CloseIcon width='20px' height='20px' fill='black' />
      ) : (
        <BurgerMenuIcon width='25px' height='16px' viewBox='0 0 25 16' fill='black' />
      )
    }
    onClick={onHandleToggleMenu}
  />
))

HamburgerMenu.displayName = 'HamburgerMenu'

export default HamburgerMenu
