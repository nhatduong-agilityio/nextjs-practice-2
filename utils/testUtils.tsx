import { ChakraProvider } from '@chakra-ui/react'
import { AuthProvider } from '@store/providers/AuthProvider'
import { LoadingProvider } from '@store/providers/LoadingProvider'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'
import { customTheme } from '@themes/index'
import type { ReactElement } from 'react'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ChakraProvider theme={customTheme}>
      <LoadingProvider>
        <AuthProvider>{children}</AuthProvider>
      </LoadingProvider>
    </ChakraProvider>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
