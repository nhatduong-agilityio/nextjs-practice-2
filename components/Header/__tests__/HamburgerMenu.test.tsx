// Utils
import { render } from '@utils/testUtils'

// Components
import HamburgerMenu from '../HamburgerMenu'

describe('HamburgerMenu render', () => {
  const propsMock = {
    toggleMenu: true,
    onHandleToggleMenu: jest.fn(),
  }

  it('Should HamburgerMenu match Snapshot', () => {
    const { container } = render(<HamburgerMenu {...propsMock} />)

    expect(container).toMatchSnapshot()
  })
})
