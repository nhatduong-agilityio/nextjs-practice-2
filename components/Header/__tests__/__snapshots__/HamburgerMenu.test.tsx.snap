// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerMenu render Should HamburgerMenu match Snapshot 1`] = `
<div>
  <button
    aria-label="Open Menu"
    class="chakra-button css-1gxx722"
    data-testid="hamburgerMenu"
    style="margin-left: 18px;"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-1tlmmfk"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
